tests/cpp/test_allocation_domain.cpp

```diff
@@ -1414,4 +1417,50 @@ TEST_F(AllocationDomainTest, ClearReductionIterDomainsPatch) {
       tv1->getContiguity(), ElementsAre(contig_copy[0], contig_copy[2]));
 }
 
+TEST_F(AllocationDomainTest, InputAllocationIsSplit_Concrete) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({6});
+  TensorView* out = set(in);
+  fusion->addInput(in);
+  fusion->addOutput(out);
+
+  in->split(0, 2);
+  in->setAllocationDomain(in->getLoopDomain(), true);
```

Comment:
out of curiosity, do we still require allocation to be on the path between root to loop? I'm wondering if this means it's now a requirement that we *need to* keep the allocation and loop domain the same?

Nitpicking on this since this goes through a scheduler and is not a manually scheduled fusion.

Reply:
> do we still require allocation to be on the path between root to loop?

No. When you see me doing that, it's likely because the right API is lacking or a workaround for a pre-existing limitation. 

For this particular case, how do I set the allocation domain without having to also make it the loop? IIUC, `in->split` automatically adds the split results to the loop domain.

Reply:
Got'ya.

`TensorView::split`, as well as `TensorDomain::split`, does update the loop domain directly. They are just using `IterDomain::split` tho, which can be called directly. 
https://github.com/NVIDIA/Fuser/blob/a86071521d81b52555e81848567caa4996b4b111/csrc/ir/nodes.cpp#L3414

Reply:
You can also use an AbstractTensor for this purpose as in https://github.com/NVIDIA/Fuser/blob/e57a755ca63e89c231ea4d114d891ffafc326099/csrc/scheduler/ampere_multi_matmul.cpp#L43

Reply:
Per https://github.com/NVIDIA/Fuser/pull/3458#discussion_r1854611891, this PR will assume that allocation equals loop **when** there's a split. This assumption is to work around a limitation in TransformReplay::replayCasP and/or the matmul scheduler which unconditionally propagates allocation domains. The assumption itself isn't ideal. A fusion/kernel input is not generated by a loop and therefore shouldn't have a loop domain to begin with.